### ISSUE TITLE
blobconverter cannot read Tensorflow model file

### ISSUE THREAD

**Issue by wiseOsprey1** [luxonis/blobconverter#12]
Problem:
I can't get blobconverter to read my Tensorflow model file.

Model file graph freeze and export built using:
python Tensorflow\models\research\object_detection\exporter_main_v2.py  --input_type=image_tensor --pipeline_config_path=Tensorflow\workspace\models\my_ssd_mobnet\pipeline.config --trained_checkpoint_dir=Tensorflow\workspace\models\my_ssd_mobnet --output_directory=Tensorflow\workspace\models\my_ssd_mobnet\export

blobconverter command:
py -m blobconverter --tensorflow-pb C:\Users\jonsc\OneDrive\Documents\PYTHON\MachineLearning\TFODCourse-main\Tensorflow\workspace\models\my_ssd_mobnet\export\saved_model\saved_model.pb --shaves 6

Errors:
Downloading C:\Users\jonsc\.cache\blobconverter\saved_model_openvino_2021.3_6shave.blob...
{
    "exit_code": 1,
    "message": "Command failed with exit code 1, command: /usr/bin/python3 model_compiler/openvino_2021.3/converter.py --precisions FP16 --output_dir /tmp/blobconverter/a642629270b8432f837a3716efaec82e --download_dir /tmp/blobconverter/a642629270b8432f837a3716efaec82e --name saved_model --model_root /tmp/blobconverter/a642629270b8432f837a3716efaec82e",
    "stderr": "[ FRAMEWORK ERROR ]  Cannot load input model: TensorFlow cannot read the model file: \"/tmp/blobconverter/a642629270b8432f837a3716efaec82e/saved_model/FP16/saved_model.pb\" is incorrect TensorFlow model file. \nThe file should contain one of the following TensorFlow graphs:\n1. frozen graph in text or binary format\n2. inference graph for freezing with checkpoint (--input_checkpoint) in text or binary format\n3. meta graph\n\nMake sure that --input_model_is_text is provided for a model in text format. By default, a model is interpreted in binary format. Framework error details: Error parsing message. \n For more information please refer to Model Optimizer FAQ, question #43. (https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html?question=43#question-43)\n",
    "stdout": "========== Converting saved_model to IR (FP16)\nConversion command: /usr/bin/python3 -- /opt/intel/openvino/deployment_tools/model_optimizer/mo.py --framework=tf --data_type=FP16 --output_dir=/tmp/blobconverter/a642629270b8432f837a3716efaec82e/saved_model/FP16 --model_name=saved_model --data_type=FP16 --input_model=/tmp/blobconverter/a642629270b8432f837a3716efaec82e/saved_model/FP16/saved_model.pb\n\nModel Optimizer arguments:\nCommon parameters:\n\t- Path to the Input Model: \t/tmp/blobconverter/a642629270b8432f837a3716efaec82e/saved_model/FP16/saved_model.pb\n\t- Path for generated IR: \t/tmp/blobconverter/a642629270b8432f837a3716efaec82e/saved_model/FP16\n\t- IR output name: \tsaved_model\n\t- Log level: \tERROR\n\t- Batch: \tNot specified, inherited from the model\n\t- Input layers: \tNot specified, inherited from the model\n\t- Output layers: \tNot specified, inherited from the model\n\t- Input shapes: \tNot specified, inherited from the model\n\t- Mean values: \tNot specified\n\t- Scale values: \tNot specified\n\t- Scale factor: \tNot specified\n\t- Precision of IR: \tFP16\n\t- Enable fusing: \tTrue\n\t- Enable grouped convolutions fusing: \tTrue\n\t- Move mean values to preprocess section: \tNone\n\t- Reverse input channels: \tFalse\nTensorFlow specific parameters:\n\t- Input model in text protobuf format: \tFalse\n\t- Path to model dump for TensorBoard: \tNone\n\t- List of shared libraries with TensorFlow custom layers implementation: \tNone\n\t- Update the configuration file with input/output node names: \tNone\n\t- Use configuration file used to generate the model with Object Detection API: \tNone\n\t- Use the config file: \tNone\n\t- Inference Engine found in: \t/opt/intel/openvino/python/python3.6/openvino\nInference Engine version: \t2.1.2021.3.0-2787-60059f2c755-releases/2021/3\nModel Optimizer version: \t    2021.3.0-2787-60059f2c755-releases/2021/3\nFAILED:\nsaved_model\n"
}

I would very much appreciate it if someone would summarize the core issues addressed in the error messages and suggest how to fix them.

**Comment by wiseOsprey1** [comment]
The source model is an inference graph for freezing with checkpoint in text format.
Any ideas why it can't be read/recognized?

**Comment by wiseOsprey1** [comment]
I tried training the model using a base CNN of "SSD MobileNet V2 320x320" instead of "SSD MobileNet V2 **FPNLite** 320x320" and the loss statistics were significantly worse.

Over 2000 steps, "SSD MobileNet V2 320x320" had an initial loss of 31.210 and then exploded to 968558.562 at step 1400.  It never converged to a decent value after that -- staying above 25 after increasing the total number of steps over multiple sequential runs.
whereas "SSD MobileNet V2 FPNLite 320x320" had an initial loss of 2.6 and final loss of 0.204.

I increased the batch size from 4 to 8 to see if that would help the loss convergence of "SSD MobileNet V2 320x320".  This dramatically improved the initial loss (was 31.21 and is now 2.348), but it stalls out at around 1.7.

So if the fpnlite version of the model isn't compatible with OpenVino, am I out of luck?

**Comment by VanDavv** [comment]
Hi @wiseOsprey1, 
Could you try exporting the model in binary format and trying again? Per suggestion [here](https://chromium.googlesource.com/external/github.com/tensorflow/tensorflow/+/r0.10/tensorflow/g3doc/how_tos/tool_developers/index.md#Text-or-Binary), binary format is size-optimized and would be preferred over text format.

You can pass `--input_model_is_text` using `--optimizer-params="--input_model_is_text"` if you want to stick with text format and see if it helps.

> So if the fpnlite version of the model isn't compatible with OpenVino, am I out of luck?
We're using OpenVINO both in blobconverter and DepthAI, so if models are not compatible with it, you won't be able to compile/run them
